lib/publiccloud/basetest.pm

```diff
@@ -114,16 +114,40 @@ sub _cleanup {
     eval { $self->cleanup(); } or bmwqemu::fctwarn("self::cleanup() failed -- $@");
 
     my $flags = $self->test_flags();
+
     # currently we have two cases when cleanup of image will be skipped:
-    # 1. Calling module needs to have publiccloud_multi_module => 1 test flag
-    # and should not have result = 'fail' and 'fatal' flag
+    # 1. Job should have 'PUBLIC_CLOUD_NO_CLEANUP' variable and result == 'fail'
+    return if ($self->{result} && $self->{result} eq 'fail' && get_var('PUBLIC_CLOUD_NO_CLEANUP_ON_FAILURE'));
```

Comment:
AFAIK this will generate perl exception nevertheless , let's do something like 
```suggestion
    return if (exists($self->{result}) && $self->{result} eq 'fail' && get_var('PUBLIC_CLOUD_NO_CLEANUP_ON_FAILURE'));
```